crates/router/src/core/routing/helpers.rs

```diff
@@ -812,6 +843,16 @@ pub async fn push_metrics_with_update_window_for_success_based_routing(
         );
         logger::debug!("successfully pushed success_based_routing metrics");
 
+        state
+            .store
+            .insert_dynamic_routing_stat_entry(dynamic_routing_stats)
+            .await
+            .map_err(|e| {
+                logger::error!(dynamic_routing_db_metrics_error=?e);
+                errors::ApiErrorResponse::InternalServerError
```

Comment:
isn't this a redundant log. you are anyways logging this functions result in parent function. you can do `change_context`